### ISSUE TITLE
How do I get train loss inside a callback?

### ISSUE THREAD

**Issue by EderSantana** [fchollet/keras#591]
I'm writing a callback to plot progress using Bokeh. I'd like to show `loss` and `valid_loss` on it. But the `logs` dict only has the later. Is there any way to get the former inside the `on_epoch_end` of a callback?

Tkx!

**Comment by fchollet** [comment]
It is passed to the `on_epoch_end` `logs` dictionary. Different callback methods get different data, and validation data only gets computed at the end of each epoch. 

You can take a look at `callbacks.BaseLogger` to see how it handles validation data display.

**Comment by EderSantana** [comment]
Thanks!!! 

I just saw how it is done in here https://github.com/fchollet/keras/blob/aa21a15bd3b1f5a32f49710d771f2c59deb9c7a0/keras/callbacks.py#L231-L256

@fchollet is there a reason for `on_epoch_end` not be getting `loss` and `loss_acc`??? The moving average could be ran inside the model.fit, so each callback wouldn't have to repeat the process.

**Comment by fchollet** [comment]
> The moving average could be ran inside the model.fit, so each callback wouldn't have to repeat the process.

Yes, this leads to redundancy in the current setup. We could definitely start computing it in `fit` (which is what we were doing before the callback system)... not much of an issue for now, though it could become one if we have lots of callbacks. In the standard case, only `BaseLogger` does this computation, so it only occurs once.

**Comment by Dylan-Cunningham** [comment]
Here is an example with something I was working on...
`class MyCallback(Callback):
    def __init__(self):
        super(MyCallback, self).__init__()
        ...
    def on_epoch_end(self, epoch, logs={}):
        results = [logs['loss'], logs['val_loss'], logs['acc'], logs['val_acc']]
        ...`
